src/x86_64/pci.c

```diff
@@ -222,10 +335,34 @@ void pci_discover()
     }
 }
 
+/*
+ * See https://wiki.osdev.org/PCI#Enumerating_PCI_Buses
+ */
+void pci_discover()
+{
+    struct pci_dev _dev = { .bus = 0, .slot = 0, .function = 0 };
+    pci_dev dev = &_dev;
+
+    if ((pci_get_hdrtype(dev) & PCIM_MFDEV) == 0) {
+        // single PCI host controller
+        pci_probe_bus(0);
+    } else {
+        // multiple PCI host controllers
+        for (int f = 1; f < 8; f++) {
```

Comment:
Is there a constant out of pci.h that could be used here?

Reply:
FIxed